pallets/resource-discussions/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+// pub use crate::weights::WeightInfo;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+// #[cfg(feature = "runtime-benchmarks")]
+// mod benchmarking;
+// pub mod weights;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{dispatch::RawOrigin, pallet_prelude::*, transactional};
+    use frame_system::pallet_prelude::*;
+    use sp_std::convert::TryInto;
+
+    use pallet_posts::{NextPostId, PostExtension};
+    use subsocial_support::{Content, PostId};
+
+    // use crate::weights::WeightInfo;
+
+    type ResourceId<T> = BoundedVec<u8, <T as Config>::MaxResourceIdLength>;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_posts::Config + pallet_spaces::Config {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// The maximum number of characters in a resource id.
+        #[pallet::constant]
+        type MaxResourceIdLength: Get<u32>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        ResourceDiscussionLinked {
+            resource_id: ResourceId<T>,
+            account_id: T::AccountId,
+            post_id: PostId,
+        },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        ResourceDiscussionAlreadyCreated,
+    }
+
+    #[pallet::storage]
+    #[pallet::getter(fn resource_post)]
+    pub type ResourceDiscussion<T: Config> =
+        StorageDoubleMap<_, Blake2_128Concat, ResourceId<T>, Twox64Concat, T::AccountId, PostId>;
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(Weight::from_ref_time(500_000_000))]
+        #[transactional]
+        pub fn link_post_to_resource(
```

Comment:
Should this be called in a similar way to `create_resource_discussion()` to be more consistent? Because right now the names of these 2 extrinsics are quite inconsistent: one is talking about discussion, while another (this one) is calling this as post.
```suggestion
        pub fn update_resource_discussion(
```